agent-control/src/agent_type/agent_attributes.rs

```diff
@@ -1,22 +1,65 @@
-use std::collections::HashMap;
+use std::{collections::HashMap, path::PathBuf};
+
+use thiserror::Error;
+use tracing::debug;
+
+use crate::agent_control::agent_id::AgentID;
 
 use super::variable::{Variable, namespace::Namespace};
 
 /// contains any attribute from the sub-agent that is used to build or modify variables used to template the AgentType.
 #[derive(Debug, PartialEq, Clone, Default)]
 pub struct AgentAttributes {
     /// sub-agent Agent ID
-    pub agent_id: String,
+    agent_id: String,
+    auto_generated_dir: PathBuf,
 }
 
+#[derive(Debug, Error)]
+#[error("Failed to create AgentAttributes: {0}")]
+pub struct AgentAttributesCreateError(String);
+
 impl AgentAttributes {
-    const VARIABLE_SUB_AGENT_ID: &'static str = "agent_id";
+    pub const VARIABLE_SUB_AGENT_ID: &'static str = "agent_id";
+    pub const GENERATED_DIR: &'static str = "agent_dir";
```

Comment:
I do not like that much this name...
IMO
```
          ${nr-sub:agent_dir}/randomdir/randomfile.yaml
```

is not that clear "autogenerate_agent_dir"? I am not sure since it is really long, but agent_dir is not descriptive enough

Reply:
Makes sense, I'll change it.